### PR TITLE
Avoid extra request to shared storage, fixed extra re-renders

### LINKED ISSUE #5845: Request /api/server/share is called when the task creation form is opened
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour

This request should only be called after opening a `connected file share` tab. This can potentially be a long request if the user has a lot of files in the share root directory. Moreover, it's a useless action if the user will upload data from other sources(computer/cloud).
image

### Your Environment

- Git hash commit (`git log -1`):
- Docker version `docker version` (e.g. Docker 17.0.05):
- Are you using Docker Swarm or Kubernetes?
- Operating System and version (e.g. Linux, Windows, MacOS):
- Code example or link to GitHub repo or gist to reproduce problem:
- Other diagnostic information / logs:
    <details>
    <summary>Logs from `cvat` container</summary>
    </details>

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Resolved #5845 

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat-ui/src/components/file-manager/file-manager.tsx [hunk 1]
```diff
@@ -156,9 +164,19 @@ export class FileManager extends React.PureComponent<Props, State> {
         }
 
         const { SHARE_MOUNT_GUIDE_URL } = config;
-        const { treeData, onUploadShareFiles, onLoadData } = this.props;
+        const {
+            treeData, sharedStorageInitialized, onUploadShareFiles, onLoadData,
+        } = this.props;
         const { expandedKeys, files } = this.state;
 
+        if (!sharedStorageInitialized) {
```
Comment (Marishka17) [inline cvat-ui/src/components/file-manager/file-manager.tsx:172]: Maybe need to move it inside and return the tab component once, what do you think?
```
<Tabs.TabPane key='share' tab='Connected file share'>
...
</Tabs.TabPane>
```

File: cvat-ui/src/reducers/share-reducer.ts [hunk 2]
```diff
@@ -20,6 +23,12 @@ export default function (
     action: ShareActions | AuthActions | BoundariesActions,
 ): ShareState {
     switch (action.type) {
+        case ShareActionTypes.LOAD_SHARE_DATA: {
+            return {
+                ...state,
+                fetching: true,
+            };
```
Comment (Marishka17) [inline cvat-ui/src/reducers/share-reducer.ts:30]:
```suggestion
            return {
                ...state,
                initialized: false,
                fetching: true,
            };
```

**Review by Marishka17 — COMMENTED** [review]
LGTM, I have only minor comments.
Looks like a small problem with spin location
image

File: cvat-ui/src/reducers/share-reducer.ts (hunk 2 above)
Comment (bsekachev) [reply cvat-ui/src/reducers/share-reducer.ts:30]: Not in this case

File: cvat-ui/src/components/file-manager/file-manager.tsx (hunk 1 above)
Comment (bsekachev) [reply cvat-ui/src/components/file-manager/file-manager.tsx]: Done

**Review by Marishka17 — APPROVED** [review]

**Pull request merged**